packages/core/helper-plugin/lib/src/components/GenericInput/index.js

```diff
@@ -127,16 +127,17 @@ const GenericInput = ({
 
   switch (type) {
     case 'bool': {
+      const isClearable = isNullable && !disabled;
```

Comment:
I think this is something that needs to be handle in the DS and not in the helper plugin as the bug might occur in some other place of the app that does not use the GenericInput.

BTW this input is made for dynamic forms but it comes with a cost since it imports the entire inputs library so we might not want to use it for simple forms.

Reply:
The DS does this in a way already: https://github.com/strapi/design-system/blob/f45f61988eab512a17a8a520e15006272e7954a7/packages/strapi-design-system/src/ToggleInput/ToggleInput.js#L44

If a `clearLabel` and `onClear` callback is passed in, the component renders the additional button. At the time we agreed, that the meaning of `isNullable` should not be moved into the DS, because it makes the component more specific to strapi core.

I don't think we are currently using nullable ToggleInput fields outside of dynamic forms, but it is certainly good to keep in mind 👍🏼

Reply:
I think we could pass disabled prop to ClearButton

```
{clearLabel && onClear && checked !== null && <ClearButton disabled={disabled} onClick={onClear}>{clearLabel}</ClearButton>}
```

We handle it mainly this way on the DS, we don't remove the button we only disable it, wdyt?

Reply:
I'll talk to @maevalienard to see what she thinks: I'm more in favor or not displaying it. Anyway - I'll move the issue into the DS and close this one then.